sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ApproximateUnique.java

```diff
@@ -394,35 +396,14 @@ public LargestUnique mergeAccumulators(Iterable<LargestUnique> heaps) {
       Iterator<LargestUnique> iterator = heaps.iterator();
       LargestUnique heap = iterator.next();
       while (iterator.hasNext()) {
-        List<Long> largestHashes = iterator.next().extractOrderedList();
-        for (long hash : largestHashes) {
-          if (!heap.add(hash)) {
-            break; // The remainder of this list is all smaller.
-          }
-        }
+        iterator.next().heap.forEach(h -> heap.add(h));
```

Comment:
It's rather confusing that "heap" refers both to PriorityQueue objects and LargestUnique objects, which have different behavior of .add() (one is limited and deduplicates, the other doesn't). In this line there's 2 things called "heap" referring to both of these. Maybe rename the variable "heap" to be called "accum"? (applies throughout the PR wherever a LargestUnique variable, field or parameter is called "heap")

Reply:
Agreed. I felt it too. will update.

Reply:
Done.